subsys/bluetooth/Kconfig

```diff
@@ -173,6 +173,39 @@ source "subsys/bluetooth/common/Kconfig"
 source "subsys/bluetooth/host/Kconfig"
 source "subsys/bluetooth/controller/Kconfig"
 source "subsys/bluetooth/shell/Kconfig"
+
+config BT_HCI_VS_CUSTOM
```

Comment:
I'm not sure this is the best name :/ Looks like Kconfig should follow a naming scheme along the lines of `BT_LL_HAS_VS_COMMANDS` instead and let `BT_HCI_VS depend on BT_LL_HAS_VS_COMMANDS`. 

Does these configs change the behavior of the Zephyr controller as well? Or are they just used by the host?

Reply:
Yes, controller also uses `BT_HCI_VS`. 

I like idea with `BT_LL_HAS_VS_COMMAND`. I think the config name has to contain `_CUSTOM_` suffix (e.g. `BT_LL_HAS_CUSTOM_VS_COMMAND`) since the intention of this config is that LL has custom VS, not the Zehpyr's one and because `BT_HCI_VS` already defines that LL has VS commands.

Reply:
> Yes, controller also uses `BT_HCI_VS`.
Ok, then this config doesn't belong in bluetooth/host :wink: 

I think we need a clear way of differentiating between the different HOST/LL combinations. Perhaps the root issue is not using something like `BT_HCI_HAS_CMD_<vendor specific command (group)>` -- though I'm not suggesting tackling that issue here, that could require some effort and requires more coordination with other NCS guys.

The point of this PR is to add support for a controller to indicate whether or not it supports the Zephyr BT host's VS commands, right? Then I suggest as a starter to use a single `*_HAS_VS_*` config.

Reply:
Changed

Reply:
I would rather rename it to `BT_HCI_VS_ENABLE`.

Reply:
The BT_HCI_VS is the (use side) option that enables the option, but the `_HAS_` symbol defines if the underlying driver supports it. I.e., the `BT_HCI_VS depends on BT_LL_HAS_VENDOR_COMMANDS`. The former is an action (enable/disable) and the latter a query.

Reply:
Ok. I get rid of two things in the proposed name: 
1. LL, because it is not related to LL
2. COMMANDS, because it is both commands and events

Reply:
1. I'm not too sure about this one. We want to have an option for the controller to say that it does not support the VS commands/events that the host supports. Is `BT_CTRL_*` better, or do you have another suggestion? It is easy to get confused about which part supports what, so I'd like the name to be as clear as possible about that.
2. This is fine :+1:

Reply:
I see. What about `BT_CTRL_SUPPORTS_VS_HCI`?

Reply:
Let me put my two cents here. If we don't follow `..._HAS_...` pattern then I think we should follow the naming style for the others configuration parameters here. So it probably should be like this:  `BT_HCI_VS_<...>`.

Reply:
`HAS` or `SUPPORTS` should both work, I see some inconsistency in the zephyr  tree already here. The important thing IMO is that it is a question :wink:

Reply:
Does `BT_HAS_HCI_VS` sound ok then?